lib/json-schema/schema/reader.rb

```diff
@@ -15,7 +14,34 @@ class ReadRefused < StandardError
       def initialize(location, type)
```

Comment:
I think this could be DRYed up by refactoring the constructor to just take the object which could not be read - `@location` is always set to `location_object.to_s` and `@type` is `:uri` if the object is a URI, `:file` otherwise.
Just saw that, not definitely sure about it. What do you think?